general/glossary.md

```diff
@@ -0,0 +1,52 @@
+# Glossary
+
+This documents aims at providing a common lexicon for terms used to refer to artifacts.
+
+### Motivation
+
+Throughout the documentation there are several terms which are often abused because we either have no alternative or there's not enough consensus on what's their canonical name.
+
+This leads to often ambiguous documentation because one term is used for different things or because we use different terms for the same artifact/object.
+
+If this document succeeds, we should end up having a documentation where each object/artifact has one single term to designate it. In other words, [Wittgenstein](https://en.wikipedia.org/wiki/Ludwig_Wittgenstein) should be proud of us (see [Tractatus Logico-Philosophicus](https://en.wikipedia.org/wiki/Tractatus_Logico-Philosophicus)).
+
+Note that we also include an [appendix section](#appendix-trade-marks--product-names) listing the canonical spelling and casing for used trade marks or product names.
+
+## Technical Terms
+
+| Term              | Description                                                                       |
+| ----------------- | --------------------------------------------------------------------------------- |
+| AMD module        | A JavaScript AMD module (the `.js` file or its logical representation at runtime) |
+| CommonJS module   | A JavaScript CommonJS module (the `.js` files used in node)                       |
+| ES6+ module       | EcmaScript 6 or higher code (i.e.: `import from` syntax)                          |
```

Comment:
I most commonly see these referred to as "ES modules"; and less frequently:

- "ECMAScript modules": eg. https://nodejs.org/api/esm.html
- Just "modules" or "JavaScript modules"; eg: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules
- "ESModules": https://kentcdodds.com/blog/super-simple-start-to-es-modules-in-the-browser/

Most people aren't referring to these by number any more ([ES6 got renamed to ES 2015 anyway](https://en.wikipedia.org/wiki/ECMAScript#6th_Edition_%E2%80%93_ECMAScript_2015)).

If you want to keep the "EcmaScript" in the description, change it to "ECMAScript".